app/services/subscription_reference_generator.rb

```diff
@@ -0,0 +1,32 @@
+class SubscriptionReferenceGenerator
+  def initialize(search_criteria:)
+    update(search_criteria)
+  end
+
+  def update(search_criteria)
```

Comment:
I'm not sure we need this method here (unless it's being used somewhere I can't see) - by convention we set instance variables in the initializer

Reply:
Yeah. I put it in because I thought I would use it but didn't. Removing it.